### ISSUE TITLE
How to optimize anchors in YOLO11?

### ISSUE THREAD

**Issue by MIGANEKAWAKO** [ultralytics/ultralytics#21459]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I know that YOLO11 is anchor-free, but I want to know whether the predicted box can be optimized in YOLO11 to make it closer to the target box? Because in my task, I observed that the box predicted by the model has a significant difference in position and size from the target box, and I want to know how to change this.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @MIGANEKAWAKO, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

---

*This is an automated response. An Ultralytics engineer will assist you here soon! 🛠️*

**Comment by Y-T-G** [comment]
What's your mAP? You probably need to train more or improve your data.

**Comment by MIGANEKAWAKO** [comment]
> What's your mAP? You probably need to train more or improve your data.

I have achieved a mAP50 of 0.873 by adding attention to YOLO11, which is the best I have been able to achieve. I also used data augmentation and trained multiple times, so I think it will be difficult to beat this by tweaking the model. I wonder if there is a way to fix the problem I am having by optimizing the anchors.

**Comment by glenn-jocher** [comment]
YOLO11 is [anchor-free](https://www.ultralytics.com/glossary/anchor-free-detectors), meaning there are no anchors to optimize - it predicts object locations directly by identifying keypoints like object centers. With mAP50 of 0.873, your model is performing well. To improve bounding box precision, consider adjusting the box loss weight in your training configuration (`box` parameter) or experimenting with different IoU loss functions, but at this performance level you're likely near the ceiling for your dataset.

**Comment by MIGANEKAWAKO** [comment]
> YOLO11 is [anchor-free](https://www.ultralytics.com/glossary/anchor-free-detectors), meaning there are no anchors to optimize - it predicts object locations directly by identifying keypoints like object centers. With mAP50 of 0.873, your model is performing well. To improve bounding box precision, consider adjusting the box loss weight in your training configuration (`box` parameter) or experimenting with different IoU loss functions, but at this performance level you're likely near the ceiling for your dataset.

Oh I found it, but how should I adjust it?

**Comment by glenn-jocher** [comment]
You can adjust the box loss weight by setting the `box` parameter in your training hyperparameters - try increasing it from the default (typically 7.5) to values like 10.0 or 15.0 to put more emphasis on bounding box regression. For IoU loss functions, you can experiment with different types by setting the `iou_type` parameter to options like 'CIoU' (default), 'DIoU', 'GIoU', or 'SIoU' in your training configuration file or directly in the train call: `model.train(data='your_data.yaml', box=10.0, iou_type='DIoU')`.

**Comment by MIGANEKAWAKO** [comment]
> You can adjust the box loss weight by setting the `box` parameter in your training hyperparameters - try increasing it from the default (typically 7.5) to values like 10.0 or 15.0 to put more emphasis on bounding box regression. For IoU loss functions, you can experiment with different types by setting the `iou_type` parameter to options like 'CIoU' (default), 'DIoU', 'GIoU', or 'SIoU' in your training configuration file or directly in the train call: `model.train(data='your_data.yaml', box=10.0, iou_type='DIoU')`.

OK, I will try that. Thanks for your help!

**Comment by glenn-jocher** [comment]
You're welcome! Let me know how the box loss and IoU type adjustments work out for your specific dataset - those tweaks should help improve your bounding box precision.

**Comment by MIGANEKAWAKO** [comment]
> You're welcome! Let me know how the box loss and IoU type adjustments work out for your specific dataset - those tweaks should help improve your bounding box precision.

Hi, I tried setting the box loss to 10.0 and got an mAP of 0.863, which is 0.01 lower than before, and the box predicted by the model is still a big gap from the ground truth. Do I need to increase the bounding box loss? (By the way, the IoU loss function I used is WIoU, which performs well on my dataset.)

**Comment by glenn-jocher** [comment]
Since increasing the box loss weight actually decreased your mAP, try reducing it instead - test values like 5.0 or 3.0 to see if that improves the balance. You might also consider that at mAP50 of 0.873, the remaining gap between predictions and ground truth could be inherent to your dataset's annotation quality or object complexity, and further loss tuning may have diminishing returns.

**Comment by MIGANEKAWAKO** [comment]
> Since increasing the box loss weight actually decreased your mAP, try reducing it instead - test values like 5.0 or 3.0 to see if that improves the balance. You might also consider that at mAP50 of 0.873, the remaining gap between predictions and ground truth could be inherent to your dataset's annotation quality or object complexity, and further loss tuning may have diminishing returns.

That sounds like a good idea. I'll try it again, thanks for the advice!

**Comment by glenn-jocher** [comment]
You're welcome! Good luck with the lower box loss values - sometimes less is more when fine-tuning at high performance levels like yours.
